shell/renderer/electron_api_service_impl.cc

```diff
@@ -54,24 +55,19 @@ void InvokeIpcCallback(v8::Local<v8::Context> context,
   if (ipcNative.IsEmpty())
     return;
 
-  // Only set up the node::CallbackScope if there's a node environment.
-  // Sandboxed renderers don't have a node environment.
-  node::Environment* env = node::Environment::GetCurrent(context);
-  std::unique_ptr<node::CallbackScope> callback_scope;
```

Comment:
I think we still need this when there is a node environment?

Reply:
I believe the node callback scope is responsible for invoking async hooks, which are totally broken in Electron currently, but shouldn't be 🙃

Reply:
The problem is not with async hooks, we can bypass those checks with `export NODE_OPTIONS="--no-force-async-hooks-checks"`, although not a ideal flow.

Here the problem is about `node::CallbackScope` triggering a microtask checkpoint at an undesired time.